### ISSUE TITLE
Bash fails on `if [...` and `if [[...` on local environment and Jenkins respectively

### ISSUE THREAD

**Issue by neomatrix369** [AdoptOpenJDK/openjdk-build#174]
I have noticed lines like this:
`if [...` fails on my local machine
and 

`if [[...` fails on Jenkins

See `sbin/common-functions.sh: line 103`, to see the block of code I'm talking about, there are a couple more such implementations.

Anyone has any idea whats happening here?
Bash version issue?

```
bash --version
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```
From my local env.

**Comment by sxa555** [comment]
That syntax looks very strange to me - basically it's:

`if [ $(bash ./configure ... && $MAKE all && $MAKE install) -ne 0 ]; then`

The result of the $(...) is going to be the whole text output, not the return code, so comparing it to zero doesn't strike me as making a lot of sense. I was going to suggest running the command in the line before then `if [ $? -ne 0 ]` ot similar but it looks like you have explicitly switched it away from that in https://github.com/AdoptOpenJDK/openjdk-build/commit/5a311d824d25989e0e9ca110829f4e52f5180a00 @neomatrix369 - what was the reasoning for that?

**Comment by gibfahn** [comment]
So maybe change:

```bash
if [ $(bash ./configure ... && $MAKE all && $MAKE install) -ne 0 ]; then
```

to

```bash
if ! $(bash ./configure ... && $MAKE all && $MAKE install); then
```

**Comment by neomatrix369** [comment]
@sxa555 happened when I was battling with local env, docker, jenkins and shellcheck - but i'll give @gibfahn approach a try and see what happens.

**Comment by neomatrix369** [comment]
Fixed via PR #175

**Comment by neomatrix369** [comment]
Thanks @sxa555 and @gibfahn for your feedback.
